src/solve.jl

```diff
@@ -1,3 +1,32 @@
+abstract type OptimizationResults end #experimental; comments welcome
+mutable struct GalacticOptimizationResults{O, Tx, Tf, Tls, Tsb} <: OptimizationResults
```

Comment:
Make it singular?

Reply:
To match the naming of our other things, I think we'd do `OptimizationSolution`. Then down the line (not right now, need to do a few things), make it a `AbstractNoTimeSolution` where the minimizer is `.u`.

Reply:
Thanks for the review. I've corrected it now but there seems to be a problem with the tests. I've checked everything before submitting and it works fine. The failure trace complains about `maxiters` in the CMAES test. I tested all the CMAE algorithms manually and they give a normal output. Could it be that the test itself is faulty?